### PR TITLE
[MODULAR] Remove Common and Overhauls Languages

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request

# Constantly Updating Doc: https://hackmd.io/@RimiNosha/HkCbe1H35
I'm not updating this post until things are actually in place, provided this one doesn't just get shut and reopened at a later date!

-snip-

## How This Contributes To The Skyrat Roleplay Experience
-snip-

## Changelog

:cl:

/:cl:

### CODE REVIEW

File: code/modules/language/beachbum.dm [hunk 1]
```diff
@@ -17,3 +17,4 @@
 					"tail", "trim", "tube", "wobble", "roll", "gnarly", "epic")
 
 	icon_state = "beach"
+	secret = TRUE // SKYRAT EDIT ADDITION - Nope.
```
Comment (tf-4) [inline code/modules/language/beachbum.dm]: modularise these core language changes

File: modular_skyrat/modules/_nocommonallowed/code/languages/sol.dm [hunk 2]
```diff
@@ -8,11 +8,12 @@
...
+		"hin", "deed", "sed", "ut", "unde", "omnis", "latire", "iste", "natus", "sit", "vol", "totam", "rem", "eaque",
+		"ipsa", "quae", "ab", "illo", "et", "quasi", "dicta", "dorime", "sunt", "enim", "ipsam", "aut", "odit", "qui",
+		"amet", "que", "eius", "modi", "inci","ad", "vel", "eum", "iure", "hic", "pa", "mit", "dis", "du",
 		"di", "tol", "mi", "solari", "ite", "domum"
 	)
 	icon_state = "solcommon"
 	icon = 'modular_skyrat/master_files/icons/misc/language.dmi'
-	default_priority = 90
+	default_priority = 100
+	category = CORE_LANGUAGE // SKYRAT EDIT ADDITION - Language overhaul
```
Comment (tf-4) [inline modular_skyrat/modules/_nocommonallowed/code/languages/sol.dm]: dont need a comment here

File: modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		/datum/language/sol,
+		/datum/language/yangyu,
+		/datum/language/panslavic
+	)
+	var/list/given_languages = list()
+	desc = "A small portable microcomputer.\n<span class=\"warning\">Required for autotranslation services.</span>"
+
+/obj/item/modular_computer/tablet/pda/dropped(mob/user, silent = FALSE)
+	. = ..()
+	if (istype(user, /mob/living/carbon/human))
```
Comment (tf-4) [inline modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm:12]: try not to nest if statements where you can feasibly early return

File: modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+/obj/item/modular_computer/tablet/pda/equipped(mob/user, slot, initial = FALSE)
+	. = ..()
+	if (istype(user, /mob/living/carbon/human))
+		give_languages(user)
+
+/obj/item/modular_computer/tablet/pda/proc/give_languages(mob/user)
+	for(var/language in autotranslated_languages)
+		if(!user.has_language(language))
+			user.grant_language(language, TRUE, FALSE)
```
Comment (tf-4) [inline modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm:24]:
```suggestion
			user.grant_language(language, understood = TRUE, spoken = FALSE)
```
name arguments where it's not self-explanatory

File: modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+			given_languages.Add(language)
+
+/obj/item/modular_computer/tablet/pda/proc/take_languages(mob/user)
```
Comment (tf-4) [inline modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm:27]:
```suggestion
/obj/item/modular_computer/tablet/pda/proc/remove_languages(mob/user)
```
Could kind of, maybe, be misunderstood as "get languages". Maybe I'm just wired different

File: modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	for(var/language in given_languages)
+		user.remove_language(language)
+	given_languages.Cut()
+
+/obj/item/modular_computer/tablet/pda/proc/pda_in_inventory(mob/user)
+	for(var/content in user.contents)
+		if(istype(content, /obj/item/modular_computer/tablet/pda))
+			return TRUE
```
Comment (tf-4) [inline modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm:35]: I'm somewhat sure there's a better way to do this but it's not coming to me right now so just putting a pin here

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm [hunk 7]
```diff
@@ -276,12 +276,12 @@ GLOBAL_LIST_EMPTY(customizable_races)
 /datum/species/human/felinid
 	mutant_bodyparts = list()
 	default_mutant_bodyparts = list("tail" = "Cat", "ears" = "Cat")
-	learnable_languages = list(/datum/language/common, /datum/language/nekomimetic)
+	learnable_languages = list(/datum/language/yangyu)
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm]: not really a fan of this, there should be a proper replacement if you want to remove a species language, especially for a species this popular. If yangyu wasn't also a core language, I'd be more fine with it.
Species languages are something I think are quite important in keeping species feeling distinct, even if people can understand what they're saying.

**Review by tf-4 — CHANGES_REQUESTED** [review]
Initial glance over. Good premise, definitely interested.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
The idea is interesting, but it needs to be elaborated on further, as it has major consequences over the flow of gameplay, so we'd like for the design to be completely fleshed out before it gets further.

**Comment by hugmeorelse** [comment]
As I vocalized in the Discord, I'm not entirely sold on this idea, as this requires further discussion and talks.

**Comment by RimiNosha** [comment]
Exactly why this is draft, and has bare minimum code for a proof-of-concept for now!

**Comment by Whygil** [comment]
Try communicating through messages in PDA for a whole round, because you don't understand your boss. Cool idea for RP project, that cuts your ability to speak to your colleagues. There is no RP reason to hire a perosnell, who could don't understand each other, there is no actual reason for that, besides just to annoy. I honestly have no idea how this didn't get closed immediately

**Comment by tf-4** [comment]
> Try communicating through messages in PDA for a whole round, because you don't understand your boss. Cool idea for RP project, that cuts your ability to speak to your colleagues. There is no RP reason to hire a perosnell, who could don't understand each other, there is no actual reason for that, besides just to annoy. I honestly have no idea how this didn't get closed immediately

You're misunderstanding. The PDA acts somewhat like a circuit translator, it grants you the ability to understand the languages as you hear them. 
You are not forced to use PDA messaging.

**Comment by RimiNosha** [comment]
Yea, please actually read the concept and watch the video on how it works before jumping to conclusions, thanks!

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm (hunk 7 above)
Comment (RimiNosha) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm]: To be undone next commit!

File: modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm (hunk 5 above)
Comment (RimiNosha) [reply modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm:27]: To be pushed

**Comment by HoodMilk** [comment]
Being given a random language is not great for RP. Imagine a Russian character not actually being able to speak it because they rolled the 1/3 chance wrong. But if you don't make the core language random the server will probably settle on one to become the new common, which makes the removal of common kind of pointless. Also seems to just be needlessly messing up ghost roles for no real reason.
The PDA working as a translator for the more common languages aboard the station is a great idea though, and seems realistic to the setting.
Nekomimetic being removed is weird considering it is a racial language.

**Comment by RimiNosha** [comment]
Oh my god, I misread, ignore that-

Anyways, I doubt folk will settle on one new common, cause I know a bunch of folk will pick Yangu or Sol at the very least, and yes, you're able to pick your core language!

Nekomimetic being removed is not gonna be a thing, and a different PR is gonna de-meme-ify the language!

**Comment by RimiNosha** [comment]
Updated the doc!

Notable changes include:

- The three core languages are now Sol, Akulan and Tajaran, to make more sense with the large governments in the galaxy!
- More documented edge cases and examples!
- Splitting my changes into three digestible parts!
- Mention bringing back backgrounds in part 3 of these changes!
- A few other things that I forgot cause I don't have a diff for the doc-

File: modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm (hunk 6 above)
Comment (RimiNosha) [reply modular_skyrat/modules/_nocommonallowed/code/pdatranslation.dm:35]: Betterified for next commit!

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm (hunk 7 above)
Comment (RimiNosha) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm]: Turns out this one's still up in the air, cause Xyel isn't a huge fan of neko being a thing, and there'll be Tajaran being a core language-

**Comment by RimiNosha** [comment]
Uhhh, linter seems to have died, and I don't *think* it's my fault-

**Comment by Lukaster42** [comment]
Uncommon is being used by HECU.

**Comment by theselfish** [comment]
I really dislike this PR.
It's making communication needlessly complicated for very little upside.

"It's more roleplay!" isn't always a positive.
As Jungle stated on another PR "More roleplay is not always good roleplay."

**Comment by Iamgoofball** [comment]
>removes common

This is a bad idea that sounds like a good idea until it goes into practice and you find out how crippled the crew is when communication is interrupted. The radio being down is already a massive blow to the crew. The radio being down *and* tower of babel's going on? Complete inability for the crew to effectively communicate during crisis.

**Comment by Iamgoofball** [comment]
I think the best way to put it as a thought experiment is that if a wizard spell gave everyone a random language, how fucked would the station be?

**Comment by RimiNosha** [comment]
@theselfish @Iamgoofball

Did you two even read the doc?

**Comment by theselfish** [comment]
> @theselfish @Iamgoofball
> 
> Did you two even read the doc?

Yes.
There are good parts in the design documents.

The core language concept is not one of them.

At worst it'll be the Tower of Babel.
At best it'll be "90% of the server using one core language while annoying people select something else just to be annoying."

They get translated. This eliminates the **point** of differing languages.
As Goof said, if Comms go down (they frequently do and a lot of our players aren't smart enough to fix them) the station, in your hypothetical Station of Babel would fall into disarray.

This would also completely fuck Casm'd miners.
"Oh you speak Greek? Guess what all the Doctors, and HoP speak Japanese!"

Languages are either for species members to talk among themselves or Antags to have secrecy. That is the mechanical purpose of Languages.

**Comment by Iamgoofball** [comment]
i don't need to read the design document to know that the game design of "people talking to eachother" doesn't need more complexity added and definitely doesn't benefit from added complexity with minimal if any depth 

if it ain't broke, don't fix it is the motto of game design, and communication currently isn't broken, so what are you fixing here?

The only previous problem with communication at Skyrat was "massive levels of OOC toxicity towards people using species languages" and I fixed that by reworking languages with tgui prefs so that Gamer McElitist the human main has no reason to whine about Vox speaking Vox because he can simply learn Vox.

People whining about metacliques every time they saw someone use a species language dropped by like 90% after I made that change

Splitting Common into three languages, randomly assigning them, and making players have to pick one is just going to result in a bunch of code that amounts to "everyone just takes Sol as the new Common" and now there's a massively overcomplicated system in place because a problem that didnt exist got "solved"

**Comment by Iamgoofball** [comment]
on top of all that, there's a ton of code that assumes common exists and is the main language, and while we can patch all of it, is it worth the nonmodular edits needed for a system that adds complexity but no depth to something that was working fine already?

**Comment by Iamgoofball** [comment]
in summary, 10 years of spaceman design has taught me that fucking with common radio and/or fucking with the spaceman lingua franca makes the spaceman experience suck, especially when it comes to communication and Skyrat as an RP server is like 85% communication based, and any changes to communication need to really consider the ramifications of said change

**Comment by RimiNosha** [comment]
> At worst it'll be the Tower of Babel. At best it'll be "90% of the server using one core language while annoying people select something else just to be annoying."

I do indeed fear that Sol would end up becoming the de-facto language, which is why I want backgrounds back to encourage at least some diversity!

In all honesty, I might try and do backgrounds in part one or two instead, and shuffle some non-essential things to part three instead-

> They get translated. This eliminates the **point** of differing languages. As Goof said, if Comms go down (they frequently do and a lot of our players aren't smart enough to fix them) the station, in your hypothetical Station of Babel would fall into disarray.

Translation isn't 100% confirmed to be attached to **NTNet** (*not* comms!) yet. If that does happen, it will almost certainly be a testmerge to see how it gets affected. NTNet doesn't go down with ion anomalies unless it's the variant that completely shuts down telecomms, which is relatively rare!

Also, with racial languages being heavily trimmed down, there'd be a solid chance of everyone in the department being able to communicate at some level. Also consider that I plan to add spacer as a "universal" interpreted language, which means it's usable by anyone, but with somewhat mixed results.

Honestly, it might an idea, with automapper being a thing now, to add in a seperate translation server machine that's in the vault, science server room or telecomms that'd work across all z-levels, with protected places like the centcom z-layers always having translation services.

> This would also completely fuck Casm'd miners. "Oh you speak Greek? Guess what all the Doctors, and HoP speak Japanese!"

Casm'd? I didn't know that randomized languages, unless that's new a thing or a special ruin-

> Languages are either for species members to talk among themselves or Antags to have secrecy. That is the mechanical purpose of Languages.

In TG, yes.

**Comment by theselfish** [comment]
> > This would also completely fuck Casm'd miners. "Oh you speak Greek? Guess what all the Doctors, and HoP speak Japanese!"
> 
> Casm'd? I didn't know that randomized languages, unless that's new a thing or a special ruin-


Shouldn't have used a Casm'd miner as example.
Someone who's just a brain, or needs any extensive treatment where they lose their things (Ashdrakes come to mind.)
If you don't speak what the HoP does, they'll understand you, but you won't understand them.


> > Languages are either for species members to talk among themselves or Antags to have secrecy. That is the mechanical purpose of Languages.
> 
 > In TG, yes.


Guess who's a TG downstream?

**Comment by RimiNosha** [comment]
> Someone who's just a brain, or needs any extensive treatment where they lose their things (Ashdrakes come to mind.)

MMIs would just universally translate, honestly. They can already sustain a brain, so why wouldn't it allow the brain to speak any of the core languages?

And medbay is perfectly capable of requesting a PDA, or using said Core Translator they have for this exact reason, which I just remembered I put in the doc.

> Guess who's a TG downstream?

Us. But that doesn't mean we have to stick to TG principles. If we did, we wouldn't have:
- Microfusion Rifles
- Guncargo
- Freely retractable modsuits
- The ghost cafe
- The (upcoming) removal of EORG
- Our airlocks
- Vanguard operatives along with our gate missions, like the black mesa one
- Our current levels of language bloat
- Custom races, along with the insane level of customization we have along with that
- Combat indicators
- The interlink
- Heretics that don't gib on sacrifice (though I think TG mighta removed this)
- Barbers
- NTRep and Blueshield
- DS2 and Tarkon
- Fleshed out ashies and Interdyne ops that aren't just punching bags for miners

And that's just the stuff off the top of my head-

**Comment by Iamgoofball** [comment]
i dont think you actually read my post because you didn't actually address the point being asked:

what is broken that you are fixing here?

**Comment by Iamgoofball** [comment]
I've read the design doc now and still don't see an answer to these two questions:
1. What problem is being fixed by removing Common?
2. What justifies such a massive change to an incredibly core mechanic of SS13? This would be on par with removing atmospherics or removing and subdividing an entire department into 3 separate departments in terms of how much of a core change this is.

Combat mode was a massive improvement to the UX of gameplay interaction.

Cloning removal allowed us to completely refactor revival and medbay to be significantly more interesting and better gameplay wise.

How is this change justified and how does it outweigh the downsides of complexity, clunkiness, and complexity without depth?

Because to be candid with you, this entire design seems like it exists for the sake of existing.

**Comment by Iamgoofball** [comment]
Oh, and in regards to your list:
# Microfusion Rifles
Added because skyrat wanted weapons that were reloadable, but needed to switch to Energy to preserve balance.
It's also a minor feature that didn't impact or change core mechanics. It's just more guns.

# Guncargo
Added because skyrat to add more content to cargo.
It's also a minor feature that didn't impact or change core mechanics. It's just more content for cargo.

# Freely retractable modsuits
Certified /tg/ classic, actually. 😎

# The ghost cafe
Interlink runs out of room sometimes.
It's also a minor feature that didn't impact or change core mechanics. It's just more rooms for stuff.

# The (upcoming) removal of EORG
It's just a rules change and a layout adjustment to CC. This is frankly minor in the scope of things.

# Our airlocks
You mean Tau Ceti's airlocks?

# Vanguard operatives along with our gate missions, like the black mesa one
This is just a new job and content fill for away missions.
It's also a minor feature that didn't impact or change core mechanics. It's just a science job(that i dislike because muh away teams) and some more maps for away missions.

# Our current levels of language bloat
People like customization. I didn't see anyone complaining about this until these PRs rolled around.
It's also a minor feature that didn't impact or change core mechanics.

# Custom races, along with the insane level of customization we have along with that
People like customization. 
Skyrat is also a roleplaying server populated by the furry community of SS13 and people want to roleplay their characters, thus making /tg/'s lackluster character customization on a vanilla /tg/ copy subpar and a problem that needs resolved. Thus, the new character creator was done.

# Combat indicators
Skyrat is a roleplaying server where the rules require people to talk things out before choosing violence. Combat Indicators, and their relevant logging, allows people to be held to these rules and standards. Their non existence was a problem that needed solved for Skyrat, It's a core mechanic change, but it's a mechanic change that needed done.

# The interlink
Due to Skyrat's allowances for player behavior via the rules, a way for station crewmembers to leave the round with other crew for extended periods of time is needed. This is a major change to a core mechanic(arrivals), but it was done carefully, precisely, and with a very specific problem that needed resolved(no safe places for player behavior).

# Heretics that don't gib on sacrifice (though I think TG mighta removed this)
Certified /tg/ classic, actually. 😎

# Barbers
It's just a new service job that cuts hair and gives backrubs.
It's also a minor feature that didn't impact or change core mechanics. It's just a new RP fluff job.

# NTRep and Blueshield
It's just two bad jobs cargo culted from citadel which cargo culted them from paradise which cargo culted them from bay.
It's also a minor feature that didn't impact or change core mechanics. It's just two jobs.

# DS2 and Tarkon
It's just two stations in space.
It's also a minor feature that didn't impact or change core mechanics. It's just two stations in space.

# Fleshed out ashies and Interdyne ops that aren't just punching bags for miners
These are, frankly, minor adjustments. We gave them a better starting base and some more tools.
It's also a minor feature that didn't impact or change core mechanics. It's just a bit of mapping and a few new items/crafting recipes.


# "ok?"
The takeaway is that all of the changes you listed are either
1. Minor changes that don't impact core game design of SS13 or it's ability to be a roleplaying game
2. Specific, targeted changes designed to fix large problems with Skyrat and it's ability to be a roleplaying platform.
3. Certified /tg/ classics 😎 

So, with all that in mind, I ask you once again:

### **What problem does this PR solve, and why is this problem so important that Common, a core mechanic, needs removed and a complex yet lacking in depth system implemented in it's place?**

And finally, to address this:

# "we don't HAVE to stick to /tg/"
Have you ever heard the phrase term, just because you can, doesn't mean you should?
We aren't dumb at /tg/. We don't make changes for no reason. We have over 15 years of iterative design on what Space Station 13 is. If you're chomping at the bit to remove /tg/'s "mistakes", maybe direct your ire towards something that *does* need a rework, like Xenobiology or Virology. And not core fundamentals like the ability to speak and have other spacemen understand you.

**Comment by RimiNosha** [comment]
Closing for overhaul, cause this thread and branch are already messy, and the direction has changed drastically.

**Pull request closed without merging**